components/brave_news/browser/feed_v2_builder.h

```diff
@@ -41,14 +46,25 @@ class FeedV2Builder {
   FeedV2Builder& operator=(const FeedV2Builder&) = delete;
   ~FeedV2Builder();
 
-  void Build(bool recalculate_signals, BuildFeedCallback callback);
+  void AddListener(mojo::PendingRemote<mojom::FeedListener> listener);
+
   void BuildChannelFeed(const std::string& channel, BuildFeedCallback callback);
   void BuildPublisherFeed(const std::string& publisher_id,
                           BuildFeedCallback callback);
+  void BuildAllFeed(BuildFeedCallback callback);
 
   void GetSignals(GetSignalsCallback callback);
 
  private:
+  using UpdateCallback = base::OnceCallback<void()>;
+  struct UpdateSettings {
+    bool signals = false;
+    bool suggested_publishers = false;
+    bool feed = false;
+    bool topics = false;
```

Comment:
I think you might want to get rid of the default copy constructor from this struct.

Reply:
If I delete the copy constructor, this is no longer an `aggregate class` in C++ 20 (as it will have a user declared constructor), so I won't be able to do the aggregate initialization. Should I move to using a constructor instead?

```
UpdateSettings(/*signals=*/true, /*suggested_publishers=*/false, /*feed=*/false, /*topics=*/false);
```

I think it reads a little bit less nicely, and is definitely a bit less pleasant to use.

https://stackoverflow.com/a/64108206/3260044

Reply:
I'm curious if you try defaulting the move constructor. I guess when you delete the copy constructor, the implicit move constructor won't be generated by the compiler. So I think this might work.

```
struct UpdateSettings {
  UpdateSettings() = default;
  UpdateSettings(UpdateSettings&) = delete;
  UpdateSettings& operator=(UpdateSettings&) = delete;
  UpdateSettings(UpdateSettings&&) = default;
  UpdateSettings& operator=(UpdateSettings&&) = default;
};
```

Reply:
Ah, I think I'm wrong, testing.

Reply:
I think it works because the toy is using C++ 14 - this was a breaking change in C++ 20. Apparently it was done to fix this:


```cpp
struct SuperSecret {
private:
   SuperSecret() = default;
};

void Main() {
    // I'm a hacker now :laughing: 
    SuperSecret foo{};
}
```

Reply:
Oh, thanks! I didn't know that the spec was changed since c++20. I didn't read but https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p1008r1.pdf this seems to be the one.